app/src/main/java/com/example/tasky/feature_login/presentation/LoginScreen.kt

```diff
@@ -136,7 +138,10 @@ fun LoginScreenContent(
 
             is AuthenticationViewState.Failure -> {
                 // Show an Alert Dialog with API failure Error code/message
-                val message = (viewState as AuthenticationViewState.Failure).message
+                val message =
+                    (viewState as AuthenticationViewState.Failure).dataError.toLoginErrorMessage(
+                        context = LocalContext.current
+                    )
```

Comment:
With the sealed class recommendation from above, this cast would not be necessary here, since you made the type check before 👍

Reply:
ohhh nice!